### PR TITLE
Integrate create and display comments

### PR DESCRIPTION
**Pull request by Pranjal-R-Agrawal**
1. Completed Display Post Use Case and added tests for the interactor and controller
2. Beautified Create Comment View and updated the tests to reflect the changes
3. Integrated Display Post Use Case and Create Comment Use Case
4. Create Comment View is now dynamically resized based on the size of the text fields
5. Implemented a tentative method to display posts using viewmanager
    i. Main creates the display post view model and view (with the create comments integration)
    ii. The view and view model are then stored in ViewManager
    iii. ViewManagerModel can then fire a property change indicating which post to show
    iv. A function to display post will be called which will then call getDisplayCommentsView
    v. The final post view will be launched in a new window

I have not added any tests for the post and comment view because there is not much to test and regardless, its structure is too complex to extract buttons etc. reliably from it due to its extremely nested and recursive structure. However, the interactor tests do check the validity of all the data that is used to create comment view.

The overall line coverage of the tests is about 70% as of this PR.

### CODE REVIEW

File: src/main/java/use_case/display_comment/application_business_rules/DisplayCommentInteractor.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            comments.add(displayCommentDataAccessObject.getCommentByCommentID(id));
+            if (comments.get(0) == null) comments.clear();
+        } else {
+            comments = displayCommentDataAccessObject.getCommentsByParentPostID(id);
+        }
+
+        DisplayCommentOutputData outputData = new DisplayCommentOutputData(comments.size());
+
+        if (!comments.isEmpty()) {
+            post = displayCommentDataAccessObject.getPostByPostID(comments.get(0).getParentPostId());
```
Comment (YathusanK) [inline src/main/java/use_case/display_comment/application_business_rules/DisplayCommentInteractor.java:40]: Why not use the id variable that was set at the start of the method?

Comment (Pranjal-R-Agrawal) [reply src/main/java/use_case/display_comment/application_business_rules/DisplayCommentInteractor.java:40]: Ah, that's actually a naming issue. Originally, the variable parentPostId stored the id of the post however, I then expanded the functionality to also fetch individual comments. This is required to update the view with the comment a user makes after the post has been loaded. Hence, the variable could actually be storing the id of a post or that of a comment, and the variable config needs to be used to determine what it actually is. Ig I just forgot to rename it.

**Comment by Pranjal-R-Agrawal** [comment]
Temporarily closing pull request since I need to make some changes

**Review by YathusanK — APPROVED** [review]
Overall your code is very well written. The code follows Clean Architecture and SOLID principles such as the Single Responsibility principle and the Dependency Inversion Principle were followed. Your integration of create_comment and display_comment also functions properly. Furthermore, you used design patterns like builders and factories.

File: src/main/java/app/CreateCommentUseCaseBuilder.java [hunk 2]
```diff
@@ -2,32 +2,42 @@
...
-    public static CreateCommentController createCreateCommentUseCase(ViewManagerModel viewManagerModel,
+    public static CreateCommentController createCreateCommentUseCase(PostAndCommentsViewModel postAndCommentsViewModel,
                                                                      CreateCommentViewModel createCommentViewModel,
                                                                      CreateCommentDataAccessInterface createCommentDataAccessObject ){
-        // TODO: Add in PostViewModel when implemented since its needed for a Presenter Method
-        CreateCommentOutputBoundary createCommentPresenter = new CreateCommentPresenter(viewManagerModel, createCommentViewModel);
+        CreateCommentOutputBoundary createCommentPresenter = new CreateCommentPresenter(postAndCommentsViewModel, createCommentViewModel);
         CreateCommentInputBoundary createCommentInteractor = new CreateCommentInteractor(createCommentPresenter, createCommentDataAccessObject, new CommentFactory());
         return new CreateCommentController(createCommentInteractor);
     }
```
Comment (anbu1504) [inline src/main/java/app/CreateCommentUseCaseBuilder.java:43]: Overall great use of the builder design pattern for structuring a use case for the general CreateCommentUseCase.

File: src/main/java/data_access/DisplayCommentDataAccessInterface.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import java.util.List;
+
+public interface DisplayCommentDataAccessInterface {
+    Comment getCommentByCommentID(ObjectId id);
+    Post getPostByPostID(ObjectId id);
+    List<Comment> getCommentsByParentPostID(ObjectId id);
+    User getUserById(ObjectId id);
+    ObjectId getLoggedInUserId();
+}
```
Comment (anbu1504) [inline src/main/java/data_access/DisplayCommentDataAccessInterface.java:16]: You did a good job of using the interface segregation principle (ISP) to ensure that you are only adding necessary methods. This also ensures that this interface does not have many dependencies and can be reused wherever needed. This also ensures a clear contract in terms of what the classes that implement this interface need to do to fulfill their implementation of this respective interface.

**Review by anbu1504 — APPROVED** [review]
This code was well written and does indeed follow the structure that Clean Architecture outlines. Furthermore, the code also ensures to uphold the SOLID principles (for instance the Interface Segregation Principle and the Single Responsibility Principle) so that any changes that need to be made can be done so in a quick and efficient manner. .

You also made sure that you stuck with the naming conventions for all variable names and class names and method names so that there is a clear communication of what each thing is. There is also the presence of tests which also cover a good amount of the code. 

Please ensure that you add the javadoc for all public methods and classes.

**Pull request merged**